### PR TITLE
Improve semantics of the SelectorGroup

### PR DESCRIPTION
**Pull request by connor-baer**
## Approach and changes

- Changed the SelectorGroup to render as a `fieldset` with a `legend`. This makes the label visible by default. It can be hidden with the `hideLabel` prop, but this should only be done in rare cases and only if the purpose of the field can be inferred from context. 
- Added a `hideLabel` prop to the RadioButtonGroup to visually hide the label. This should only be used in rare cases and only if the purpose of the field can be inferred from context.

## Definition of done

* [x] Development completed
* [x] Reviewers assigned
* [x] Unit and integration tests
* [x] Meets minimum browser support
* [x] Meets accessibility requirements

### CODE REVIEW

**Review by amelako — APPROVED** [review]
🌟

File: packages/circuit-ui/components/RadioButtonGroup/RadioButtonGroup.tsx [hunk 1]
```diff
@@ -86,14 +99,15 @@ export const RadioButtonGroup = forwardRef(
...
+      hideLabel,
       ...props
     }: RadioButtonGroupProps,
     ref: RadioButtonGroupProps['ref'],
   ) => {
     const name = customName || uniqueId('radio-button-group_');
     return (
       <fieldset name={name} ref={ref} {...props}>
-        {label && <Legend>{label}</Legend>}
+        {label && <Legend hideLabel={hideLabel}>{label}</Legend>}
```
Comment (robinmetral) [inline packages/circuit-ui/components/RadioButtonGroup/RadioButtonGroup.tsx:110]: I don't think we have to check. for `label` here, it's required 🙂

File: packages/circuit-ui/components/SelectorGroup/SelectorGroup.tsx [hunk 2]
```diff
@@ -122,42 +135,42 @@ export const SelectorGroup = forwardRef(
...
-                }
-                {...optionRest}
-              >
-                {children}
-              </StyledSelector>
-            </OptionItem>
-          ))}
-      </StyledSelectorGroup>
+      <Fieldset ref={ref} stretch={stretch} {...rest}>
+        {label && <Legend hideLabel={hideLabel}>{label}</Legend>}
```
Comment (robinmetral) [inline packages/circuit-ui/components/SelectorGroup/SelectorGroup.tsx:151]: Same

(this is the places where we should throw as discussed, but let's maybe do this in v4 or something)

File: packages/circuit-ui/components/RadioButtonGroup/RadioButtonGroup.tsx (hunk 1 above)
Comment (connor-baer) [reply packages/circuit-ui/components/RadioButtonGroup/RadioButtonGroup.tsx:110]: In TypeScript, but not JavaScript 🙃

Comment (robinmetral) [reply packages/circuit-ui/components/RadioButtonGroup/RadioButtonGroup.tsx:110]: Should we make this consistent?

https://github.com/sumup-oss/circuit-ui/blob/1869c21810c10182e6ace038e096c0e1d68a489c/packages/circuit-ui/components/Input/Input.tsx#L344

In the Input I assumed that _not_ passing a label would make the UI weird even if the IDE doesn't show errors, so developers would pass it (cf. what we saw in Account in the dashboard yesterday)

**Comment by robinmetral** [comment]
@connor-baer oops, did we not say that we also wanted the SelectorGroup label to be any `ReactNode`? I could fix this in a follow-up PR

**Comment by connor-baer** [comment]
@robinmetral I don't remember but that sounds like a good idea.

**Pull request merged**